src/lib/components/Bootstrap/Bootstrap.js

```diff
@@ -12,22 +12,36 @@ import PropTypes from 'prop-types';
 export default class Bootstrap extends Component {
   constructor(props) {
     super(props);
-
+    this.app = props.app;
     this.searchOnInit = props.searchOnInit;
     this.historyListen = props.historyListen;
+    this.eventListenerEnabled = props.eventListenerEnabled;
     this.onAppInitialized = props.onAppInitialized;
     this.onBrowserHistoryExternallyChanged =
       props.onBrowserHistoryExternallyChanged;
     this.searchOnUrlQueryStringChanged = props.searchOnUrlQueryStringChanged;
   }
 
+  updateQueryState = query => this.props.updateQueryState(query);
+
+  onQueryChanged = ({ detail: payload }) => {
+    let appReceiver = payload.app || this.app;
+    if (appReceiver === this.app) {
+      this.updateQueryState(payload.search_query);
+    }
```

Comment:
can you add a `else { console.debug("RSK {this.app} ignore event for the app {appReceiver}...")`

Reply:
Nice idea!